module-services/service-cellular/ServiceCellular.cpp

```diff
@@ -1737,22 +1737,63 @@ auto ServiceCellular::handleCellularAnswerIncomingCallMessage(CellularMessage *m
     return std::make_shared<CellularResponseMessage>(ret);
 }
 
+namespace
+{
+    constexpr auto translate(at::Result::Code code) -> CellularCallRequestGeneralError::ErrorType
+    {
+        switch (code) {
+        case at::Result::Code::ERROR:
+        case at::Result::Code::CME_ERROR:
+        case at::Result::Code::CMS_ERROR:
+            return CellularCallRequestGeneralError::ErrorType::Error;
+        case at::Result::Code::TIMEOUT:
+            return CellularCallRequestGeneralError::ErrorType::ModemTimeout;
+        case at::Result::Code::TOKENS:
+        case at::Result::Code::PARSING_ERROR:
+        case at::Result::Code::FULL_MSG_BUFFER:
+        case at::Result::Code::TRANSMISSION_NOT_STARTED:
+        case at::Result::Code::RECEIVING_NOT_STARTED:
+        case at::Result::Code::DATA_NOT_USED:
+        case at::Result::Code::CMUX_FRAME_ERROR:
+            return CellularCallRequestGeneralError::ErrorType::TransmissionError;
+        case at::Result::Code::OK:
+        case at::Result::Code::NONE:
+        case at::Result::Code::UNDEFINED:
+            return CellularCallRequestGeneralError::ErrorType::UndefinedError;
+        }
+        return CellularCallRequestGeneralError::ErrorType::UndefinedError;
```

Comment:
will be nice to add `[[fallthrough]]`

Reply:
fallthrough was introduced to solve slightly different issue - https://en.cppreference.com/w/cpp/language/attributes/fallthrough

Reply:
For me it is also the case here. 
`Indicates that the fall through from the previous case label is intentional and should not be diagnosed by a compiler that warns on fallthrough. `

Reply:
it only matters if there is some instruction between the labels, otherwise it is simply redundant. 
If there is an instruction between labels compiler will generate error. In case without instruction there is no warning.

Please check the example for the c++ reference manual.